packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -144,16 +151,14 @@ class OnsetPicker extends React.Component {
         const ageValues = OnsetUtils.onsetToAge(this.props.birthdate, prevState.onsetDate);
 
         return {
+          granularity,
           age: ageValues.age,
           ageUnit: ageValues.ageUnit,
           onsetDate: moment(this.props.birthdate).add(ageValues.age, ageValues.ageUnit),
         };
-      }, () => this.handleDateChange(this.state.onsetDate));
-    }
-    this.setState({ granularity });
-
-    if (this.props.granularitySelectOnChange) {
-      this.props.granularitySelectOnChange(granularity);
+      }, () => this.handleOnsetUpdate());
+    } else {
+      this.setState({ granularity }, () => this.handleOnsetUpdate());
```

Comment:
You could remove one function instantiation per execution if you provide handleOnsetUpdate directly as the callback: `this.setState({ granularity }, this.handleOnsetUpdate);`

Reply:
Ah of course, updated. I think I like the look of these little anonymous functions too much.